runners/google-cloud-dataflow-java/build.gradle

```diff
@@ -151,33 +161,34 @@ def commonExcludeCategories = [
 
 // For the following test tasks using legacy worker, set workerHarnessContainerImage to empty to
 // make Dataflow pick up the non-versioned container image, which handles a staged worker jar.
-task validatesRunnerLegacyWorkerTest(type: Test) {
-  group = "Verification"
-  dependsOn ":runners:google-cloud-dataflow-java:worker:legacy-worker:shadowJar"
-
-  systemProperty "beamTestPipelineOptions", JsonOutput.toJson([
-          "--runner=TestDataflowRunner",
-          "--project=${dataflowProject}",
-          "--region=${dataflowRegion}",
-          "--tempRoot=${dataflowValidatesTempRoot}",
-          "--dataflowWorkerJar=${dataflowLegacyWorkerJar}",
-          "--workerHarnessContainerImage=",
-  ])
-
-  // Increase test parallelism up to the number of Gradle workers. By default this is equal
-  // to the number of CPU cores, but can be increased by setting --max-workers=N.
-  maxParallelForks Integer.MAX_VALUE
-  classpath = configurations.validatesRunner
-  testClassesDirs = files(project(":sdks:java:core").sourceSets.test.output.classesDirs) +
-          files(project(project.path).sourceSets.test.output.classesDirs)
-  useJUnit {
-    includeCategories 'org.apache.beam.sdk.testing.ValidatesRunner'
-    commonExcludeCategories.each {
-      excludeCategories it
+def createValidatesRunnerTest = { name, pipelineOptions=legacyPipelineOptions, disabledTests=[] ->
```

Comment:
Why change the name to drop `LegacyWorker`? If the plan is to reuse this for other runner versions should the `worker:legacy-worker:shadowJar` dependency be injected somehow as well?

Reply:
The actual task name is the same, and defined on line 238. Here I have just made a function for producing variants. The new variant is defined on line 244.

Reply:
I meant the name of this method specifically, not the tasks. This method has a dependsOn dependency with the legacy worker, so it will always be for the legacy worker and the naming change seems to make this less clear. Unless of course, in the future, this method could be used for new runners (e.g. runner v2)?

I was considering if it would possible to pull out the `dependsOn ...legacy-worker:shadowJar` (by passing it in as a parameter, specifying the dependency in some other scope, or some other injection mechanism) and then reusing the `createValidatesRunnerTest` method for the runner v2 by passing in `pipelineOptions + ["--experiments=use_runner_v2"]`.

Reply:
Agree with your naming point, and pushed a rename.

I believe that is possible to continue to configure with a block but Groovy's scope configuration blocks confuse the heck out of me so I did not try to do it. I expect you can do it either by passing the closure in to the `createValidatesRunner` function or by returning something that is ready to accept a block. I'm a bit hazy on how/when the latter works.